pkg/trace/agent/agent.go

```diff
@@ -184,6 +186,17 @@ func (a *Agent) Process(p *api.Payload, sublayerCalculator *stats.SublayerCalcul
 			continue
 		}
 
+		// filter by tags
+		// check the keys of traces and see if they match required keys
+		// list of forbidden tags which will match tags of spans that should be dropped
+		// check if keys are present in tag of spans
+		if !a.Validator.Validates(root) {
```

Comment:
Might be nice to have `Validates` return `error` instead,  where the error returned is the reason the trace was rejected, in the case that it doesn't validate.

That would be helpful for the Debug message. Otherwise it may be difficult to understand why the trace is rejected.